plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestModule.java

```diff
@@ -20,31 +20,11 @@
 package org.elasticsearch.plugin.ingest;
 
 import org.elasticsearch.common.inject.AbstractModule;
-import org.elasticsearch.common.inject.multibindings.MapBinder;
-import org.elasticsearch.ingest.processor.append.AppendProcessor;
-import org.elasticsearch.ingest.processor.convert.ConvertProcessor;
-import org.elasticsearch.ingest.processor.date.DateProcessor;
-import org.elasticsearch.ingest.processor.fail.FailProcessor;
-import org.elasticsearch.ingest.processor.geoip.GeoIpProcessor;
-import org.elasticsearch.ingest.processor.grok.GrokProcessor;
-import org.elasticsearch.ingest.processor.gsub.GsubProcessor;
-import org.elasticsearch.ingest.processor.join.JoinProcessor;
-import org.elasticsearch.ingest.processor.lowercase.LowercaseProcessor;
-import org.elasticsearch.ingest.processor.remove.RemoveProcessor;
-import org.elasticsearch.ingest.processor.rename.RenameProcessor;
-import org.elasticsearch.ingest.processor.set.SetProcessor;
-import org.elasticsearch.ingest.processor.split.SplitProcessor;
-import org.elasticsearch.ingest.processor.trim.TrimProcessor;
-import org.elasticsearch.ingest.processor.uppercase.UppercaseProcessor;
 import org.elasticsearch.plugin.ingest.rest.IngestRestFilter;
 
-import java.util.HashMap;
-import java.util.Map;
-
 public class IngestModule extends AbstractModule {
 
     private final boolean ingestEnabled;
-    private final Map<String, ProcessorFactoryProvider> processorFactoryProviders = new HashMap<>();
 
     public IngestModule(boolean ingestEnabled) {
         this.ingestEnabled = ingestEnabled;
```

Comment:
> Moving the entire ingest infrastructure to core, but no processor implementation feels wrong to me, because core won't hold any processor implementations.

This is exactly what we have done with scripts, and it follows along with the goal of "small core", and isolation of extra dependencies/etc.  Having multiple modules with various processors doesn't matter, since they are all there by default, but the advantage of isolating these, for example when it brings in a bunch of dependencies, is huge. I only see one custom security permission, but IMO that should be a separate module/plugin from the "defaults". In either case, the abstraction being in core, and particular implementations being in modules/plugins, seems like the right thing to me.